### ISSUE TITLE
ca65: correct line flagged with error

### ISSUE THREAD

**Issue by kugelfuhr** [cc65/cc65#1634]
A missing factor in an expression causes an expected but missing token to be skipped, leading to invalid flagged errors in the following line:
```
l = 3 +
lda     #$00
```
An error should be output for line 1, but not for line 2. Actually, both are flagged as errors:
```
test.s(1): Error: Syntax error
test.s(2): Error: Unexpected trailing garbage characters
```
This can fixed as follows:
```diff
diff --git a/src/ca65/expr.c b/src/ca65/expr.c
index 8703b2a55..125760b36 100644
--- a/src/ca65/expr.c
+++ b/src/ca65/expr.c
@@ -1209,11 +1209,11 @@ static ExprNode* Factor (void)
                 SB_GetLen (&CurTok.SVal) == 1) {
                 /* A character constant */
                 N = GenLiteralExpr (TgtTranslateChar (SB_At (&CurTok.SVal, 0)));
+                NextTok ();
             } else {
                 N = GenLiteral0 ();     /* Dummy */
                 Error ("Syntax error");
             }
-            NextTok ();
             break;
     }
     return N;
```

**Comment by polluks** [comment]
You are welcome to make a PR :-)

**Comment by kugelfuhr** [comment]
Thank you. I don't have much spare time, so it might be better if someone with more experience with the process does this.

**Comment by mrdudz** [comment]
> Thank you. I don't have much spare time, so it might be better if someone with more experience with the process does this.

- clone the repo to your github account
- create a branch
- apply your fixes to the branch
- push the branch
- create a PR from the branch

The big advantage is that this way the automatic checks will get triggered for your branch

**Comment by mrdudz** [comment]
fixed via #1671
